ens/main.py

```diff
@@ -129,6 +144,61 @@ def setup_address(self, name, address=default, transact={}):
         resolver = self._set_resolver(name, transact=transact)
         return resolver.functions.setAddr(raw_name_to_hash(name), address).transact(transact)
 
+    @dict_copy
+    def setup_content(self, name, content, transact={}):
+        """
+        Set up the name to store the supplied content record.
+        The sender of the transaction must own the name, or
+        its parent name.
+
+        :param str name: ENS name to set up
+        :param dict content: content to set up, with `type` and `hash` in content hash format.
+            If resolver does not support EIP 1577, `type` is ignored and `hash` is set as HEX value.
+            If ``{}``, erase the record.
+        :param dict transact: the transaction configuration, like in
+            :meth:`~web3.eth.Eth.sendTransaction`
+        :raises InvalidName: if ``name`` has invalid syntax
+        :raises UnauthorizedError: if ``'from'`` in `transact` does not own `name`
+        """
+        owner = self.setup_owner(name, transact=transact)
+        self._assert_control(owner, name)
+
+        current = self.content(name)
+        if current == content or (not current and not content):
+            return None
+
+        transact['from'] = owner
+        resolver = self._set_resolver(name, transact=transact)
+
+        is_eip1577 = resolver.functions.supportsInterface('0xbc1c58d1').call()
```

Comment:
@pipermerriam Ok, I will move them into constants.

I will also link where are they defined but I have a little problem here. Interface for EIP 1577 is defined in [ENS resolver docs](https://docs.ens.domains/contract-api-reference/publicresolver#get-content-hash). But I don't know where a legacy interface is defined. It is 100% correct because it [was used](https://github.com/ensdomains/resolvers/commit/950a625f9352b8041a68d62adbcbca4c1bf6f2fb#diff-6cc402ed233a180ea1eae9dcd56c9051L13) in old resolver contract but I haven't saw and EIP about it? What should I do here?